File: app/src/main/java/io/github/howiezuo/unsplash/ViewModelHolder.java
package io.github.howiezuo.unsplash;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;


public class ViewModelHolder<VM> extends Fragment {

    private VM viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setRetainInstance(true);
    }

    public static <M> ViewModelHolder createHolder(M viewModel) {
        ViewModelHolder<M> holder = new ViewModelHolder<>();
        holder.setViewModel(viewModel);
        return holder;
    }

    public VM getViewModel() {
        return viewModel;
    }

    public void setViewModel(VM viewModel) {
        this.viewModel = viewModel;
    }
}

File: app/src/main/java/io/github/howiezuo/unsplash/app/ViewModelHolder.kt
package io.github.howiezuo.unsplash.app

import android.os.Bundle
import android.support.v4.app.Fragment


class ViewModelHolder<VM> : Fragment() {

    var viewModel: VM? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        retainInstance = true
    }

    companion object {
        fun <M> createHolder(vm: M): ViewModelHolder<M> {
            val holder = ViewModelHolder<M>()
            holder.viewModel = vm
            return holder
        }
    }

}
